packages/buidler-core/src/internal/buidler-evm/provider/node.ts

```diff
@@ -1172,4 +1356,56 @@ export class BuidlerNode {
       await this._stateManager.setStateRoot(initialStateRoot);
     }
   }
+
+  private _removeLogs(blockNumber: number) {
+    this._filters.forEach(filter => {
+      if (filter.type !== Type.LOGS_SUBSCRIPTION) {
+        return;
+      }
+
+      for (let i = filter.logs.length - 1; i >= 0; i--) {
+        if (
+          new BN(toBuffer(filter.logs[i].blockNumber!)).toNumber() <=
+          blockNumber
+        ) {
+          break;
+        }
+
+        filter.logs[i].removed = true;
+      }
+    });
+  }
+
+  private async _computeFilterParams(
+    filterParams: FilterParams,
+    isFilter: boolean
+  ): Promise<FilterParams> {
+    if (
+      filterParams.fromBlock === LATEST_BLOCK ||
+      filterParams.toBlock === LATEST_BLOCK
+    ) {
+      const block = await this.getLatestBlock();
+      if (filterParams.fromBlock === LATEST_BLOCK) {
+        filterParams.fromBlock = bufferToInt(block.header.number);
+      }
+
+      if (!isFilter && filterParams.toBlock === LATEST_BLOCK) {
+        filterParams.toBlock = bufferToInt(block.header.number);
+      }
+    }
+
+    return filterParams;
+  }
+
+  private _newDeadline(): Date {
+    const dt = new Date();
+    dt.setMinutes(dt.getMinutes() + 5);
```

Comment:
I think this can overflow. You can do `new Date(Date.now() + 5 * 60 * 1000)`

Reply:
It won't overflow, I'll add a comment since it's easy to come to the conclusion that it can.